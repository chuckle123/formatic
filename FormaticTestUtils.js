import React from 'react';
import { mount } from 'enzyme';

import Formatic from './lib/formatic';

export const renderFieldsToHtml = fields => {
  const config = Formatic.createConfig();
  const value = config.createRootValue({
    fields,
  });

  const component = mount(
    <Formatic
      config={config}
      value={value}
      fields={fields}
      onChange={() => {}}
    />
  );
  const html = component.html();
  component.unmount();
  return html;
};
