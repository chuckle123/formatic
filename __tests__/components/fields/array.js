/*global describe, it, expect*/

import { renderFieldsToHtml } from '../../../FormaticTestUtils';
import arrayExample from '../../../demo/examples/array';

describe('array field', () => {
  it('should render correctly', () => {
    expect(renderFieldsToHtml(arrayExample.fields)).toMatchSnapshot();
  });
});
