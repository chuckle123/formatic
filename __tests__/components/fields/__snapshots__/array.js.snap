// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array field should render correctly 1`] = `

<div class="formatic">
  <div class="optional">
    <div class>
      <span class="required-label not-required-text">
      </span>
    </div>
    <span>
      <span>
      </span>
      <fieldset class>
        <div class="optional">
          <div class>
            <label>
              Array
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div class
                 tabindex="0"
            >
              <span>
              </span>
              <div class>
                <span role="button"
                      tabindex="0"
                      class
                >
                  [add]
                </span>
              </div>
            </div>
          </span>
        </div>
        <div class="required">
          <div class>
            <label>
              Robots (R)
            </label>
            <span class="required-label required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div class
                 tabindex="0"
            >
              <span>
              </span>
              <div class>
                <select class>
                  <option value="0">
                    Simple
                  </option>
                  <option value="1">
                    Complex
                  </option>
                </select>
                <span role="button"
                      tabindex="0"
                      class
                >
                  [add]
                </span>
              </div>
            </div>
          </span>
        </div>
        <div class="optional">
          <div class>
            <label>
              Listing Pretty Text
            </label>
            <span class="required-label not-required-text">
            </span>
          </div>
          <span>
            <span>
            </span>
            <div class
                 tabindex="0"
            >
              <span>
              </span>
              <div class>
                <span role="button"
                      tabindex="0"
                      class
                >
                  [add]
                </span>
              </div>
            </div>
          </span>
        </div>
      </fieldset>
    </span>
  </div>
</div>

`;
